packages/ptokens-assets-evm/src/ptokens-evm-provider.ts

```diff
@@ -205,24 +197,37 @@ export class pTokensEvmProvider implements pTokensAssetProvider {
         return false
       }
     }, _pollingTime)
+    if (!receipt) throw new Error('Polling function stopped unexpectedly. No receipt found')
     return receipt.transactionHash.toString()
   }
 
+  /**
+   * Wait for tx receipt.
+   * @param _txHash - Transcation hash of the searched receipt.
+   * @returns TransactionReceipt of _txHash.
+   */
+  async waitForTransactionReceipt(_txHash: `0x${string}`): Promise<TransactionReceipt> {
+    const receipt: TransactionReceipt = await this._publicClient.waitForTransactionReceipt({ hash: _txHash})
+    if (!receipt) throw new Error('No receipt found')
+    return receipt
+  }
+
   protected async _pollForHubOperation(
     _hubAddress: string,
     _eventName: EVENT_NAMES,
     _fromBlock: bigint,
     _operationId: string,
     _pollingTime = 1000,
-  ) {
-    let log
+  ): Promise<Log> {
+    let log: Log | undefined
     // eslint-disable-next-line @typescript-eslint/no-unsafe-call
     await polling(async () => {
       try {
+        log = undefined
         const logs = await this._publicClient.getLogs({
           fromBlock: _fromBlock,
           address: stringUtils.addHexPrefix(_hubAddress),
-          events: events,
+          events: pNetworkHubAbi.filter((_fun) => _fun.type === 'event' && _fun.name === _eventName.toString()),
```

Comment:
events is already defined. reuse it

Reply:
events was created manually starting from pNetworkHubAbi. This means that in case of any change to pNetworkHubAbi events should be manually updated. In this way instead only EVENT_NAMES needs changes.